test/core/transport/chaotic_good/client_transport_test.cc

```diff
@@ -0,0 +1,179 @@
+using testing::Return;
+using testing::StrictMock;
+
+namespace grpc_core {
+namespace chaotic_good {
+namespace testing {
+
+class MockEndpoint
+    : public grpc_event_engine::experimental::EventEngine::Endpoint {
+ public:
+  MOCK_METHOD(
+      bool, Read,
+      (absl::AnyInvocable<void(absl::Status)> on_read,
+       grpc_event_engine::experimental::SliceBuffer* buffer,
+       const grpc_event_engine::experimental::EventEngine::Endpoint::ReadArgs*
+           args),
+      (override));
+
+  MOCK_METHOD(
+      bool, Write,
+      (absl::AnyInvocable<void(absl::Status)> on_writable,
+       grpc_event_engine::experimental::SliceBuffer* data,
+       const grpc_event_engine::experimental::EventEngine::Endpoint::WriteArgs*
+           args),
+      (override));
+
+  MOCK_METHOD(
+      const grpc_event_engine::experimental::EventEngine::ResolvedAddress&,
+      GetPeerAddress, (), (const, override));
+  MOCK_METHOD(
+      const grpc_event_engine::experimental::EventEngine::ResolvedAddress&,
+      GetLocalAddress, (), (const, override));
+};
+
+class ClientTransportTest : public ::testing::Test {
+ public:
+  ClientTransportTest()
+      : control_endpoint_ptr_(new StrictMock<MockEndpoint>()),
+        data_endpoint_ptr_(new StrictMock<MockEndpoint>()),
+        memory_allocator_(
+            ResourceQuota::Default()->memory_quota()->CreateMemoryAllocator(
+                "test")),
+        control_endpoint_(*control_endpoint_ptr_),
+        data_endpoint_(*data_endpoint_ptr_),
+        event_engine_(std::make_shared<
+                      grpc_event_engine::experimental::FuzzingEventEngine>(
+            []() {
+              grpc_timer_manager_set_threading(false);
+              grpc_event_engine::experimental::FuzzingEventEngine::Options
+                  options;
+              return options;
+            }(),
+            fuzzing_event_engine::Actions())),
+        client_transport_(
+            std::make_unique<PromiseEndpoint>(
+                std::unique_ptr<MockEndpoint>(control_endpoint_ptr_),
+                SliceBuffer()),
+            std::make_unique<PromiseEndpoint>(
+                std::unique_ptr<MockEndpoint>(data_endpoint_ptr_),
+                SliceBuffer()),
+            std::static_pointer_cast<
+                grpc_event_engine::experimental::EventEngine>(event_engine_)),
+        arena_(MakeScopedArena(initial_arena_size, &memory_allocator_)),
+        pipe_client_to_server_messages_(arena_.get()) {}
+
+ private:
+  MockEndpoint* control_endpoint_ptr_;
+  MockEndpoint* data_endpoint_ptr_;
+  size_t initial_arena_size = 1024;
+  MemoryAllocator memory_allocator_;
+
+ protected:
+  MockEndpoint& control_endpoint_;
+  MockEndpoint& data_endpoint_;
+  std::shared_ptr<grpc_event_engine::experimental::FuzzingEventEngine>
+      event_engine_;
+  ClientTransport client_transport_;
+  ScopedArenaPtr arena_;
+  Pipe<MessageHandle> pipe_client_to_server_messages_;
+};
+
+TEST_F(ClientTransportTest, AddOneStream) {
```

Comment:
Is it possible to generalize this test to add N streams and send K messages per stream ?

Reply:
Yes, I've added the tests. Please let me know if there's other test coverage needed.